aws/resource_aws_sagemaker_endpoint_configuration.go

```diff
@@ -255,3 +363,151 @@ func flattenProductionVariants(list []*sagemaker.ProductionVariant) []map[string
+	}
+
+	if v, ok := m["enable_capture"]; ok {
+		c.EnableCapture = aws.Bool(v.(bool))
+	}
+
+	if v, ok := m["kms_key_id"]; ok && v.(string) != "" {
+		c.KmsKeyId = aws.String(v.(string))
+	}
+
+	if v, ok := m["capture_content_type_header"]; ok && (len(v.([]interface{})) > 0) {
+		c.CaptureContentTypeHeader = expandSagemakerCaptureContentTypeHeader(v.([]interface{})[0].(map[string]interface{}))
+	}
+
+	return c
+}
+
+func flattenSagemakerDataCaptureConfig(dataCaptureConfig *sagemaker.DataCaptureConfig) []map[string]interface{} {
+	if dataCaptureConfig == nil {
+		return []map[string]interface{}{}
+	}
+
+	cfg := map[string]interface{}{
+		//"enable_capture":              aws.BoolValue(dataCaptureConfig.EnableCapture),
+		"initial_sampling_percentage": aws.Int64Value(dataCaptureConfig.InitialSamplingPercentage),
+		"destination_url":             aws.StringValue(dataCaptureConfig.DestinationS3Uri),
+		//"kms_key_id":                  aws.StringValue(dataCaptureConfig.KmsKeyId),
+		"capture_options": flattenSagemakerCaptureOptions(dataCaptureConfig.CaptureOptions),
+		//"capture_content_type_header": flattenSagemakerCaptureContentTypeHeader(dataCaptureConfig.CaptureContentTypeHeader),
+	}
+
+	if dataCaptureConfig.EnableCapture != nil {
+		cfg["enable_capture"] = aws.BoolValue(dataCaptureConfig.EnableCapture)
+	}
+
+	if dataCaptureConfig.KmsKeyId != nil {
+		cfg["kms_key_id"] = aws.StringValue(dataCaptureConfig.KmsKeyId)
+	}
+
+	if dataCaptureConfig.CaptureContentTypeHeader != nil {
+		cfg["capture_content_type_header"] = flattenSagemakerCaptureContentTypeHeader(dataCaptureConfig.CaptureContentTypeHeader)
+	}
+
+	return []map[string]interface{}{cfg}
+}
+
+func expandSagemakerCaptureOptions(configured []interface{}) []*sagemaker.CaptureOption {
+	containers := make([]*sagemaker.CaptureOption, 0, len(configured))
+
+	for _, lRaw := range configured {
+		data := lRaw.(map[string]interface{})
+
+		l := &sagemaker.CaptureOption{
+			CaptureMode: aws.String(data["capture_mode"].(string)),
+		}
+		containers = append(containers, l)
+	}
+
+	return containers
+}
+
+func flattenSagemakerCaptureOptions(list []*sagemaker.CaptureOption) []map[string]interface{} {
+	containers := make([]map[string]interface{}, 0, len(list))
+
+	for _, lRaw := range list {
+		captureOption := make(map[string]interface{})
+		captureOption["capture_mode"] = aws.StringValue(lRaw.CaptureMode)
+		containers = append(containers, captureOption)
+	}
+
+	return containers
+}
+
+func expandSagemakerCaptureContentTypeHeader(m map[string]interface{}) *sagemaker.CaptureContentTypeHeader {
+	c := &sagemaker.CaptureContentTypeHeader{}
+
+	if v, ok := m["csv_content_types"]; ok && len(v.([]interface{})) > 0 {
+		contentTypes := make([]*string, 0, len(v.([]interface{})))
+
+		for _, raw := range v.([]interface{}) {
+			sRaw := raw.(map[string]interface{})["content_type"]
+			contentTypes = append(contentTypes, aws.String(sRaw.(string)))
+		}
+
+		c.CsvContentTypes = contentTypes
+	}
```

Comment:
if the above changes to type are made this can be reduced to:

`c.CsvContentTypes = expandStringSet(v.(*Schema.Set))`